web_poet/page_inputs/http.py

```diff
@@ -18,13 +19,64 @@
 _AnyStrDict = Dict[AnyStr, Union[AnyStr, List[AnyStr], Tuple[AnyStr, ...]]]
 
 
-class ResponseURL(str):
-    """ URL of the response """
+class _Url:
+    def __init__(self, url: Union[str, yarl.URL], encoded=True):
```

Comment:
`url: Union[str, yarl.URL]` - this type annotation doesn't allow to create RequestUrl from RequestUrl or from ResponseUrl

Reply:
Nice catch! Fixed it in https://github.com/scrapinghub/web-poet/pull/45/commits/d869024bd2db2c5f32faf56afbb0f93502316d2a.

Reply:
I forgot to annotate the other methods in other modules. In any case, we'd be rebasing to #42 if we should proceed with this route. This would include the annotations like https://github.com/scrapinghub/web-poet/pull/42/commits/5b0e889f4800b1313877fb60e37406889e4ec413#diff-e91005d8dfc7a61689963aafad8b742554243e973f4943ec394cab425bbb61f6R81.